nestkernel/source_table.cpp

```diff
@@ -400,7 +424,11 @@ nest::SourceTable::get_next_target_data( const thread tid,
     // set the source rank
     source_rank = kernel().mpi_manager.get_process_id_of_node_id( current_source.get_node_id() );
 
-    populate_target_data_fields_( current_position, current_source, next_target_data );
+    if ( not populate_target_data_fields_( current_position, current_source, next_target_data ) )
```

Comment:
this construction is not great, and should maybe be avoided when preparing for merge into master